src/containers/Experiment/SamplesTable.js

```diff
@@ -284,19 +284,58 @@ function ThComponent({ toggleSort, className, children, ...rest }) {
   );
 }
 
-function ProcessingInformationCell({ original: sample, ...props }) {
-  let { annotations } = sample;
+// Pipeline names ref: https://github.com/AlexsLemonade/refinebio-frontend/issues/22#issuecomment-394010812
+const PIPELINES = {
+  SubmitterProcessed: 'Submitter-processed',
+  AffymetrixSCAN: 'Affymetrix SCAN',
+  Salmontools: 'Salmontools',
+  AgilentSCANTwoColor: 'Agilent SCAN TwoColor',
+  IlluminaSCAN: 'Illumina SCAN',
+  tximport: 'tximport',
+  Salmon: 'Salmon',
+  MultiQC: 'MultiQC'
+};
 
-  // If the sample is marked with `is_ccdl = false` then this sample is "Submitter Processed"
-  if (annotations.length > 0 && !annotations[0].is_ccdl) {
+function ProcessingInformationCell({ original: sample, ...props }) {
+  let { pipelines } = sample;
+
+  // Logic to decide which pipeline modal dialog should be displayed. On Keytar Kurt we're only supporting 4 types of
+  // pipelines. In the future when we add more, we might want to refactor these modal dialogs
+  if (pipelines.length === 1 && pipelines[0] === PIPELINES.AffymetrixSCAN) {
+    // check for `pipelines === ['Affymetrix SCAN']`
+    return <ScanModal sample={sample} scanType={PIPELINES.AffymetrixSCAN} />;
+  } else if (
+    pipelines.length === 1 &&
+    pipelines[0] === PIPELINES.IlluminaSCAN
+  ) {
+    // check for `pipelines === ['Illumina SCAN']`
+    return <ScanModal sample={sample} scanType={PIPELINES.IlluminaSCAN} />;
+  } else if (
+    pipelines.length === 1 &&
+    pipelines[0] === PIPELINES.SubmitterProcessed
+  ) {
+    // check for `pipelines === ['Submitter-processed']`
     return <SumitterProcessedModal sample={sample} />;
+  } else if (
+    pipelines.length === 2 &&
+    pipelines[0] === PIPELINES.tximport &&
+    pipelines[1] === PIPELINES.Salmon
+  ) {
+    // check for `pipelines === ['tximport', 'Salmon']`
+    return <SalmonTxtimportModal sample={sample} />;
+  } else if (
+    pipelines.length === 2 &&
+    pipelines[0] === PIPELINES.Salmon &&
+    pipelines[1] === PIPELINES.tximport
+  ) {
+    // check for `pipelines === ['Salmon', 'tximport']`
+    return <SalmonTxtimportModal sample={sample} />;
```

Comment:
Correct me if I'm wrong, but the checks for lines 319-325 and 326-332 seem to result in the same thing and the only thing different in the conditionals is the order of the pipelines? Am I understanding that correctly?

Reply:
Yes that's correct. I added it like that trying to make an emphasis that we're testing those two values specifically pipelines === ['tximport', 'Salmon'] or pipelines === ['Salmon', 'tximport'].

Do you think we should put these conditions together??

Reply:
Yeah, in that case, I think we might be able to simplify the checks in this function. Maybe something like this:

```
if (pipelines.length) {
   	if (pipelines.includes(PIPELINE_NAME) {
		return <AssociatedComponent />;

Reply:
hmm but with `includes` we wouldn't be checking for exact values of the `pipelines` array... and we want to ensure it only works for the combinations we defined. What do you think of this approach ??

```
  const pipelineString = pipelines.join(', ');
  
  const pipelineFactory = {
    'Affymetrix SCAN': () => <ScanModal sample={sample} scanType={PIPELINES.AffymetrixSCAN} />,
    'Illumina SCAN': () => <ScanModal sample={sample} scanType={PIPELINES.IlluminaSCAN} />,
    'Submitter-processed': () => <SumitterProcessedModal sample={sample} />,
    'tximport, Salmon': () => <SalmonTxtimportModal sample={sample} />,
    'Salmon, tximport': () => <SalmonTxtimportModal sample={sample} />,
  };

  if (pipelineFactory[pipelineString]) {
    return pipelineFactory[pipelineString]();
  } else {
    return <div>{pipelineString}</div>;
  }

```